lightning/src/routing/router.rs

```diff
@@ -2260,7 +2338,33 @@ where L::Target: Logger {
 	for results_vec in selected_paths {
 		let mut hops = Vec::with_capacity(results_vec.len());
 		for res in results_vec { hops.push(res?); }
-		paths.push(Path { hops, blinded_tail: None });
+
+		let blinded_path = payment_params.payee.blinded_route_hints().iter().enumerate()
+			.find(|(idx, (_, p))| {
+				DUMMY_BLINDED_SCID + *idx as u64 == hops[hops.len() - 1].short_channel_id ||
+					(p.blinded_hops.len() == 1 && p.introduction_node_id == hops[hops.len() - 1].pubkey)
+			})
+			.map(|(_, (_, p))| p.clone());
+		let blinded_tail = if let Some(BlindedPath { blinded_hops, blinding_point, .. }) = blinded_path {
+			let num_blinded_hops = blinded_hops.len();
+			Some(BlindedTail {
```

Comment:
nit: I think you could probably take this block forming the `BlindedTail` and put it into this previous `map` which might make it a little cleaner